common/app/views/support/HtmlCleaner.scala

```diff
@@ -795,3 +797,52 @@ object GarnettQuoteCleaner extends HtmlCleaner {
     document
   }
 }
+
+case class AffiliateLinksCleaner(pageUrl: String, sectionId: String, showAffiliateLinks: Option[Boolean], contentType: String, appendDisclaimer: Boolean = true) extends HtmlCleaner with Logging {
+
+  override def clean(document: Document): Document = {
+    if (AffiliateLinks.isSwitchedOn && AffiliateLinksCleaner.shouldAddAffiliateLinks(AffiliateLinkSections.isSwitchedOn, sectionId, showAffiliateLinks)) {
+      val cleanedResult = AffiliateLinksCleaner.replaceLinksInHtml(document, pageUrl)
+      if (cleanedResult.affiliateLinksAdded && appendDisclaimer) {
+        AffiliateLinksCleaner.insertAffiliateDisclaimer(cleanedResult.document, contentType)
+      } else {
+        cleanedResult.document
+      }
+    } else document
+  }
+}
+
+case class AffiliateLinksCleanerResult(document: Document, affiliateLinksAdded: Boolean)
```

Comment:
I could be missing a few things here (is Document mutable or copied?) but a simpler interface might be just an object here. E.g.

    object AffiliateLinksCleaner extends HtmlCleaner with Logging {
      override def clean(document: Document): Document = ???
    
      private[this] def insertAffiliateDisclaimer(document: Document, contentType: String): Document = ???
      private[this] def toSkimLink(link: String, pageUrl: String): String = ???
      private[this] def shouldAddAffiliateLinks(switchedOn: Boolean, section: String, showAffiliateLinks: Option[Boolean]): Boolean = ???
    }

That way, you avoid adding a case class for the result, as well as a case class and object for the cleaning. In addition, inlining the logic in the clean method avoids some awkward naming (for example - `val skimLinksReplaced` appears a predicate but actually mutates the document).